pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -458,9 +515,17 @@ func handleCreate(ctxArg interface{}, key string,
 			// FIXME - We really need a top level flag to tell the app is
 			//  a container. Deriving it from Storage seems hacky.
 			status.IsContainer = true
+			ss.HasResolverRef = true
+			publishAppInstanceStatus(ctx, &status)
+			resolveConfig := types.ResolveConfig{
+				DatastoreID: sc.DatastoreID,
+				Name:        sc.Name,
+				AllowNonFreePort: types.AllowNonFreePort(*ctx.globalConfig,
+					types.AppImgObj),
+				Counter: config.PurgeCmd.Counter,
+			}
+			publishResolveConfig(ctx, &resolveConfig)
```

Comment:
Why not do this work in doUpdate and not have code in both handleCreate and handleModify? Especially since the code in handleCreate and handleModify is very different.

Reply:
ok